pkg/jobqueue/dbjobqueue/dbjobqueue.go

```diff
@@ -337,6 +338,11 @@ func (q *DBJobQueue) Dequeue(ctx context.Context, jobTypes []string, channels []
 		}
 	}
 
+	// The job has been dequeued, we are past the point of no return.
```

Comment:
Doesn't this need a timeout? Otherwise it's possible to get stuck here.

Reply:
I usually dislike random timeouts to be honest. I wonder if putting one here is really necessary. It would just fail the HTTP request if the database is extremely slow for some reason. Is it a realistic scenario, though?